### ISSUE TITLE
it focuses on only 1 action per consideration, even learning is short term.

### ISSUE THREAD

**Issue by Cheeseskates** [leomwu/agario-bot#19]
a major problem with this bot is it doesn't comprehend multiple amounts of things going on.

example of this is it tries to get a blob, and a big blob is behind the bot, the bot will run away from the big blob and ignore the to be acquired small blob.

another example is when it is trying to evade. there are 2 big blobs left and right, and the two big blobs charge in to the bot. the bot will keep going back and forth, basically, shuffling, instead of going up or down away from the danger. this is because the closest big blob is only considered, and when it gets close to another big blob, it considers that, and goes back to where it was before, and repeats until one big blob goes down and changes the bots path, or when it is eaten.

this also happens occasionally when it tries to decide what small blob to eat, it will shuffle back and forth until it gets close to one from motion physics, which takes 3-10 seconds most of the time.

the chasing feature is excellent. it predicts the path the small blob goes, and will inevitably be eaten. however, if there is a small blob, or big group of small blobs, it will quit chase, even if it is just about to eat the smaller one, say, back in the corner, it will disband and go for other pellets instead. for this problem i suggest a select feature, where you click on a player, or blob, or anywhere, and it will direct itself to it, while avoiding dangers as well, as the danger might be the blob you clicked, it will deny the selection.

lastly, for the learning mode. i may be wrong about this but from my experience and testing it seems about right in my eyes. it isn't really learning mode as it says. it is just new considerations made from how it was killed last time. virus to big blob, big blob collision etc. while it would be fantastic, as i stated, it only learns from last time, not from 100 deaths as it claims. if it needs to learn, more considerations need to me made, or it actually learns in long term and not short term.

thanks for reading and good luck with the development with the bot :D

**Comment by dainiusb** [comment]
I think a feature to interrupt actions would help in this case.
Staying away from big blobs should be #1 priority.

**Comment by LeoWuVinci** [comment]
I revisit this issue from time to time but haven't really found a solution I like yet.  (I want elegance :3)

**Comment by Cheeseskates** [comment]
good luck fixing it. or someone else willing to help :D

**Comment by eaglgenes101** [comment]
Yes... it should combine all factors into one "aggregate factor" rather than picking and choosing which factor it thinks is most important.

**Comment by OverlordQ** [comment]
When it chooses to eat food aka 'Chase nearest Smaller Blob' it quite often twitches between things on the completely other side of the screen and ignores those right next to it. So it just sits in the same spot not doing anything. Or it can decide 'Hey I want to eat that food over there', where 'over there' is on the other side of a blob, so it'll approach the blob, doesn't want to get eaten, backs off, goes 'hey i want that food' and tries to move through the blob again, rinse, repeat until it gets eaten.

**Comment by LeoWuVinci** [comment]
There's too many issues in one.  This needs to be split up
